tools/cbmc/proofs/HTTP/global_state_HTTP.c

```diff
@@ -0,0 +1,201 @@
+#define MAX_DATA_SIZE 1000
+
+/* Implementation of safe malloc which returns NULL if the reuested size is 0.*/
+void *safeMalloc(size_t xWantedSize) {
+  if(xWantedSize == 0) {
+    return NULL;
+  }
+  uint8_t byte;
+  return byte ? malloc(xWantedSize) : NULL;
+}
+
+/* This modeling is required because otherwise CBMC takes
+too much time if assigned to minimumBufferSizes. */
```

Comment:
I would add a bit more explanation.  "It is common for a buffer to contain a header struct followed by user data.  We optimize CBMC performance by allocating space for the buffer using a struct with two members: the first member is the header struct and the second member is the user data.  This is faster than just allocating a sequence of bytes large enough to hold the header struct and the user data."  And format this (or make it clear) that this comment applies to the next three structs.